src/graph/immutable_graph.cc

```diff
@@ -467,53 +474,177 @@ std::vector<IdArray> ImmutableGraph::GetAdj(bool transpose, const std::string &f
   }
 }
 
-ImmutableGraph ImmutableGraph::ToImmutable(const GraphInterface* graph) {
-  const ImmutableGraph* ig = dynamic_cast<const ImmutableGraph*>(graph);
+ImGraphPtr ImmutableGraph::CreateFromCSR(
+    IdArray indptr, IdArray indices, IdArray edge_ids, const std::string &edge_dir) {
+    CSRPtr csr(new CSR(indptr, indices, edge_ids));
```

Comment:
Better to use `make_shared` instead of `new`?

Reply:
Haha, I also tried that. I found that it does not work for private constructor like this (https://github.com/dmlc/dgl/blob/master/include/dgl/immutable_graph.h#L1033).